cmd/osquery-perf/agent.go

```diff
@@ -777,28 +777,31 @@ func (a *agent) runLoop(i int, onlyAlreadyEnrolled bool) {
 		// check if we have any scheduled queries that should be returning results
 		var results []resultLog
 		now := time.Now().Unix()
-		a.scheduledQueriesMu.Lock()
 		prevCount := a.countBuffered()
-		for queryName, query := range a.scheduledQueryData {
+
+		a.scheduledQueryData.Range(func(key, value any) bool {
+			queryName := key.(string)
+			query := value.(scheduledQuery)
+
 			if query.lastRun == 0 || now >= (query.lastRun+int64(query.ScheduleInterval)) {
 				results = append(results, resultLog{
 					packName:  query.packName,
 					queryName: query.Name,
 					numRows:   int(query.numRows),
 				})
 				// Update lastRun
-				v := a.scheduledQueryData[queryName]
-				v.lastRun = now
-				a.scheduledQueryData[queryName] = v
+				query.lastRun = now
+				a.scheduledQueryData.Store(queryName, query)
```

Comment:
This still might end up with a map with inconsitent values, right? (if this gets executed at same time as the assignment to the map on the other goroutine).

Could we have a mutex to do mutual exclusion between the "logger" and "config" goroutines? (to simplify and reduce risk of races/invalid-data)

Reply:
I took the reference to the scheduledQueryData from the agent before iterating over it. If you still think I should add a muted after I can